lib/nerves/utils/http_client.ex

```diff
@@ -69,14 +69,19 @@ defmodule Nerves.Utils.HTTPClient do
     ]
 
     http_opts =
-      [timeout: :infinity, autoredirect: false] ++
-        Nerves.Utils.Proxy.config(url) ++ Keyword.get(opts, :http_opts, [])
+      [timeout: :infinity, autoredirect: false]
+      |> Keyword.merge(Nerves.Utils.Proxy.config(url))
+      |> Keyword.merge(Keyword.get(opts, :http_opts, []))
 
     opts = [stream: :self, receiver: self(), sync: false]
     :httpc.request(:get, {String.to_charlist(url), headers}, http_opts, opts, :nerves)
     {:noreply, %{s | url: url, caller: from, opts: opts, progress?: progress?}}
   end
 
+  def handle_info({:http, {_ref, {:error, {:failed_connect, [{:to_address, _address}, _reason]}}}}, s) do
+    GenServer.reply(s.caller, {:error, :failed_connect})
+  end
```

Comment:
```suggestion
  def handle_info({:http, {_ref, {:error, {:failed_connect, _}} = err}}, s) do
    GenServer.reply(s.caller, err)
  end
```
Maybe we should report back more in the error so the user can have a bit more details about the connection failure?

Reply:
re-worked as suggested :)